graph_creator/models.py

```diff
@@ -3,23 +3,124 @@
     """
 
-    active_students_amount = models.IntegerField(default=0)
+    active_students_amount_day = models.IntegerField(default=0)
+    active_students_amount_week = models.IntegerField(default=0)
+    active_students_amount_month = models.IntegerField(default=0)
     courses_amount = models.IntegerField(default=0)
-    last_data_update = models.DateTimeField(default=datetime.now())
+    data_update = models.DateTimeField(default=datetime.datetime.now())
     latitude = models.FloatField(default=0, blank=True)
     longitude = models.FloatField(default=0, blank=True)
     platform_name = models.CharField(max_length=255, null=True, blank=True)
     platform_url = models.URLField(null=True, blank=True)
     secret_token = models.CharField(max_length=255, null=True)
     statistics_level = models.CharField(max_length=255)
     students_per_country = models.TextField()
+
+    @classmethod
+    def timeline(cls):
+        """
+        Provide timeline in days for plotting on x axis.
+
+        Future: add weeks, or month for dynamic range on plots.
+        """
+        timeline_datetimes = cls.objects.order_by(
+            'data_update'
+        ).values_list('data_update', flat=True).distinct()
+
+        timeline_dates = [x.date().strftime('%Y-%m-%d') for x in timeline_datetimes]
+        return timeline_dates
+
+    @classmethod
+    def data_per_period(cls):
+        """
+        Provide total students, courses and instances, from all services per period, day by default.
+
+        We summarize values per day, because in same day we can receive data from multiple different instances.
+        We suppose, that every instance send data only once per day.
+
+        Future: add weeks, month for dynamic range on plots.
+
+        PostgreSQL: extra(select = {"month": '''DATE_TRUNC('month', creation_date)'''}) 
+        SQlite3: extra(select={'date': 'django_date_trunc("day", "testapp_log"."datetime")'})
+        """
+
+        subquery = cls.objects.order_by('data_update').annotate(
+            date_in_days=Trunc('data_update', 'day', output_field=DateField())
+        ).values('date_in_days')
+
+        students_per_day = subquery.annotate(
+            students=Sum('active_students_amount_day')
+        ).values_list('students', flat=True)
+
+        courses_per_day = subquery.annotate(courses=Sum('courses_amount')).values_list('courses', flat=True)
+        instances_per_day = subquery.annotate(instances=Count('secret_token')).values_list('instances', flat=True)
+
+        return list(students_per_day), list(courses_per_day), list(instances_per_day)
+
+    @classmethod
+    def overall_counts(cls):
+        """
+        Provide total count of all instances, courses and students from all instances per previous calendar day.
+        """
+
+        start_of_day, end_of_day = get_previous_day_start_and_end_dates()
+
+        all_unique_instances = DataStorage.objects.filter(data_update__gt=start_of_day, data_update__lt=end_of_day)
```

Comment:
Same thing about `__gte` & `__lte`. Be sure to check you're not missing any data.